ui/transactions_page.go

```diff
@@ -0,0 +1,277 @@
+								page.filterDirection[index].Layout(gtx, page.filterDirectionW)
+							})
+						})
+				}),
+			)
+		})
+	}
+}
+
+func (page *transactionsPage) txnRowHeader(common *pageCommon) {
+	gtx := common.gtx
+	txt := common.theme.Label(unit.Dp(txsRowLabelSize), "#")
+	txt.Color = common.theme.Color.Hint
+
+	layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+		layout.Flexed(rowDirectionWidth, func() {
+			txt.Layout(gtx)
+		}),
+		layout.Flexed(rowDateWidth, func() {
+			txt.Alignment = text.Middle
+			txt.Text = "Date (UTC)"
+			txt.Layout(gtx)
+		}),
+		layout.Flexed(rowStatusWidth, func() {
+			txt.Text = "Status"
+			txt.Layout(gtx)
+		}),
+		layout.Flexed(rowAmountWidth, func() {
+			txt.Text = "Amount"
+			txt.Layout(gtx)
+		}),
+		layout.Flexed(rowFeeWidth, func() {
+			txt.Text = "Fee"
+			txt.Layout(gtx)
+		}),
+	)
+}
+
+func (page *transactionsPage) txnRowInfo(common *pageCommon, transaction wallet.Transaction) {
+	gtx := common.gtx
+	txnWidgets := transactionWdg{}
+	initTxnWidgets(common, &transaction, &txnWidgets)
+
+	layout.Inset{Bottom: unit.Dp(txsPageInsetBottom)}.Layout(gtx, func() {
+		layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+			layout.Flexed(rowDirectionWidth, func() {
+				layout.Inset{Top: unit.Dp(3)}.Layout(gtx, func() {
+					txnWidgets.direction.Layout(gtx, unit.Dp(16))
+				})
+			}),
+			layout.Flexed(rowDateWidth, func() {
+				txnWidgets.time.Alignment = text.Middle
+				txnWidgets.time.Layout(gtx)
+			}),
+			layout.Flexed(rowStatusWidth, func() {
+				txt := common.theme.Body1(transaction.Status)
+				txt.Alignment = text.Middle
+				txt.Layout(gtx)
+			}),
+			layout.Flexed(rowAmountWidth, func() {
+				txnWidgets.amount.Alignment = text.End
+				txnWidgets.amount.Layout(gtx)
+			}),
+			layout.Flexed(rowFeeWidth, func() {
+				txt := common.theme.Body1(dcrutil.Amount(transaction.Txn.Fee).String())
+				txt.Alignment = text.End
+				txt.Layout(gtx)
+			}),
+		)
+	})
+}
+
+func (page *transactionsPage) Handle(common pageCommon) {
+	if page.filterSorter != page.filterSortW.Value(common.gtx) {
+		page.filterSorter = page.filterSortW.Value(common.gtx)
+		page.sortTransactions(&common)
+	}
+}
+
+func initTxnWidgets(common *pageCommon,
+	transaction *wallet.Transaction, txWidgets *transactionWdg) {
+	txWidgets.amount = common.theme.Label(unit.Dp(16), transaction.Balance)
+	txWidgets.time = common.theme.Body1("Pending")
+
+	if transaction.Status == "confirmed" {
+		txWidgets.time.Text = dcrlibwallet.ExtractDateOrTime(transaction.Txn.Timestamp)
+		txWidgets.status, _ = decredmaterial.NewIcon(icons.ActionCheckCircle)
+		txWidgets.status.Color = common.theme.Color.Success
+	} else {
+		txWidgets.status, _ = decredmaterial.NewIcon(icons.ToggleRadioButtonUnchecked)
+	}
+
+	if transaction.Txn.Direction == dcrlibwallet.TxDirectionSent {
+		txWidgets.direction, _ = decredmaterial.NewIcon(icons.ContentRemove)
+		txWidgets.direction.Color = common.theme.Color.Danger
```

Comment:
Icons should be initalized once in the addPages method. By initializing in the layout method, it's re-initalized on every frame event.